### PR TITLE
fix: use tempdir when testing static files

### LINKED ISSUE #6824: Use `tempdir` when testing static files, so temporary directories get dropped automatically
### Describe the feature

https://github.com/paradigmxyz/reth/blob/181883c97221d4ba1eccc0c83f6e3a717a08451d/testing/ef-tests/src/cases/blockchain_test.rs#L162-L165

Here and in other places where `create_test_static_files_dir` is used, we need to return both a guard (`TempDir`) and an actual path to the directory, so the guard gets dropped when goes out of scope and the directory gets deleted.

### Additional context

Blocked by https://github.com/paradigmxyz/reth/pull/6444

### PR DESCRIPTION
**Pull request by AbnerZheng**
close https://github.com/paradigmxyz/reth/issues/6824

### CODE REVIEW

File: crates/storage/provider/src/test_utils/mod.rs [hunk 1]
```diff
@@ -26,7 +26,8 @@ pub fn create_test_provider_factory() -> ProviderFactory<Arc<TempDatabase<Databa
 pub fn create_test_provider_factory_with_chain_spec(
     chain_spec: Arc<ChainSpec>,
 ) -> ProviderFactory<Arc<TempDatabase<DatabaseEnv>>> {
+    let (static_dir, _) = create_test_static_files_dir();
     let db = create_test_rw_db();
-    ProviderFactory::new(db, chain_spec, create_test_static_files_dir())
+    ProviderFactory::new(db, chain_spec, static_dir.into_path())
```
Comment (AbnerZheng) [inline crates/storage/provider/src/test_utils/mod.rs:31]: @shekhirin Tests running with this methods still can not delete the `temp_dir` by itself, Would it be beneficial to create a class, perhaps named TestProviderFactory, that owns a temp_dir? This way, when the provider is dropped, the temp_dir would also be deleted.

Comment (AbnerZheng) [reply crates/storage/provider/src/test_utils/mod.rs:31]: Good suggestion, I will apply the required change in tomorrow.

File: crates/stages/src/test_utils/test_db.rs [hunk 2]
```diff
@@ -21,34 +21,37 @@ use reth_provider::{
     HistoryWriter, ProviderError, ProviderFactory,
 };
 use std::{collections::BTreeMap, path::Path, sync::Arc};
+use tempfile::TempDir;
 
 /// Test database that is used for testing stage implementations.
 #[derive(Debug)]
 pub struct TestStageDB {
     pub factory: ProviderFactory<Arc<TempDatabase<DatabaseEnv>>>,
+    pub temp_files_dir: TempDir,
```
[the remark this answers is not in the document]
Comment (AbnerZheng) [reply crates/stages/src/test_utils/test_db.rs]: Done.

**Review by shekhirin — APPROVED** [review]
LGTM, thank you!

**Pull request merged**